slice/internal/controller/workload_controller.go

```diff
@@ -147,13 +208,54 @@ func (r *WorkloadReconciler) createSliceIfNotExist(ctx context.Context, wl *kueu
 		return err
 	}
 
-	return r.client.Create(ctx, slice)
+	log := ctrl.LoggerFrom(ctx).WithValues("slice", klog.KObj(slice))
+
+	err = r.client.Create(ctx, slice)
+	if err != nil {
+		log.Error(err, "Error creating Slice")
+		return r.handleError(ctx, wl, ac, fmt.Sprintf("Error creating Slice %q: %v", slice.Name, err), err)
+	}
+
+	msg := fmt.Sprintf("Created Slice: %q", slice.Name)
+	log.V(5).Info("Created Slice")
+	r.record.Eventf(wl, corev1.EventTypeNormal, "SliceCreated", msg)
+
+	return err
+}
+
+func (r *WorkloadReconciler) updateWorkloadAdmissionCheckStatus(ctx context.Context, wl *kueue.Workload, ac *kueue.AdmissionCheckState) error {
+	wlPatch := workload.BaseSSAWorkload(wl)
+	workload.SetAdmissionCheckState(&wlPatch.Status.AdmissionChecks, *ac, r.clock)
+	err := r.client.Status().Patch(ctx, wlPatch, client.Apply, client.FieldOwner(SliceControllerName), client.ForceOwnership)
+	if err != nil && !apierrors.IsNotFound(err) {
+		ctrl.LoggerFrom(ctx).Error(err, "Failed to patch Workload admission status")
+	}
+	return err
+}
+
+func (r *WorkloadReconciler) handleError(ctx context.Context, wl *kueue.Workload, ac *kueue.AdmissionCheckState, msg string, err error) error {
+	r.record.Event(wl, corev1.EventTypeWarning, "FailedCreateSlice", api.TruncateEventMessage(msg))
+	ac.Message = api.TruncateConditionMessage(msg)
+	patchErr := r.updateWorkloadAdmissionCheckStatus(ctx, wl, ac)
+	return errors.Join(err, patchErr)
+}
+
+func (r *WorkloadReconciler) handleSuccess(ctx context.Context, wl *kueue.Workload, ac *kueue.AdmissionCheckState) error {
+	originalState := ac.State
+	ac.State = kueue.CheckStateReady
+	ac.Message = SliceIsCreatedMessage
+	err := r.updateWorkloadAdmissionCheckStatus(ctx, wl, ac)
+	if err == nil {
+		message := fmt.Sprintf("Admission check %q updated state from %q to %q", ac.Name, originalState, kueue.CheckStateReady)
+		r.record.Event(wl, corev1.EventTypeNormal, "AdmissionCheckUpdated", message)
+	}
+	return err
 }
```

Comment:
is this a better fit in AC controller? After its creation, I think that modifications to AdmissionCheck should be done in that controller (arguably even the creation of the AdmissionCheck, but I'm less sure there)